### ISSUE TITLE
XEP-0256 support: Last Activity in Presence

### ISSUE THREAD

**Issue by mremond** [processone/ejabberd#470]
This XEP make mod_last more useful with auto away features supported by XMPP clients.

Reference:
http://xmpp.org/extensions/xep-0256.html

**Comment by manuel-rubio** [comment]
@mremond was it implemented in current ejabberd?

**Comment by badlop** [comment]
[XEP-0256](http://xmpp.org/extensions/xep-0256.html) is deprecated, and apparently only supported by the Profanity client.

Furthermore, it was superseded by [XEP-0319](https://xmpp.org/extensions/xep-0319.html) which is implemented only by Gajim, no other client or server. In fact, looking at the XEP, it doesn't require server implementation.

**Comment by weiss** [comment]
> Furthermore, it was superseded by XEP-0319 which is implemented only by Gajim, no other client or server.

I think it's implemented at least by [Conversations][1], [Monal][2], and [Swift][3] as well, by the way.

> In fact, looking at the XEP, it doesn't require server implementation.

Indeed.

[1]: https://conversations.im
[2]: https://monal.im
[3]: https://swift.im/swift.html

**Comment by badlop** [comment]
Umm, Conversations has a XEP list, but doesn't mention XEP-0256 in it. Monal had a list of XEPs but [removed it](https://github.com/monal-im/Monal/commit/8734977cf75e3066898587e775016b26a9e51227) and I couldn't find any other list. Swift has almost no documentation, or I couldn't find it.

**Comment by weiss** [comment]
> Conversations has a XEP list, but doesn't mention XEP-0256 in it.

Ah, I was referring XEP-0319.

**Comment by Jieiku** [comment]
> In fact, looking at the XEP, it doesn't require server implementation.

So that means that the clients would make use of XEP-0319 without the server needing any extra mod?

**Comment by badlop** [comment]
Right, clients can implement that XEP, and no need for the server to do anything special, just let the presence stanzas go as usual.

In this sense, a server "does implement" the XEP in the sense that clients can use it, and at the same time "does not implement" the XEP in the sense that no code in the server deals with that XEP.
